samples/adk-sql-agent/main.py

```diff
@@ -0,0 +1,135 @@
+# Example allowed origins for CORS
+ALLOWED_ORIGINS = ["http://localhost", "http://localhost:8080", "*"]
+# Set web=True if you intend to serve a web interface, False otherwise
+SERVE_WEB_INTERFACE = True
+
+logging.getLogger("opentelemetry").setLevel(logging.DEBUG)
+
+
+# [START opentelemetry_adk_otel_setup]
+def setup_opentelemetry() -> None:
+    credentials, project_id = google.auth.default()
+    resource = Resource.create(
+        attributes={
+            SERVICE_NAME: "adk-sql-agent",
+            # The project to send spans to
+            "gcp.project_id": project_id,
+        }
+    )
+
+    # Set up OTLP auth
+    request = google.auth.transport.requests.Request()
+    auth_metadata_plugin = AuthMetadataPlugin(credentials=credentials, request=request)
+    channel_creds = grpc.composite_channel_credentials(
+        grpc.ssl_channel_credentials(),
+        grpc.metadata_call_credentials(auth_metadata_plugin),
+    )
+
+    # Set up OpenTelemetry Python SDK
+    tracer_provider = TracerProvider(resource=resource)
+    tracer_provider.add_span_processor(
+        BatchSpanProcessor(
+            OTLPSpanExporter(
+                credentials=channel_creds,
+                endpoint="https://telemetry.googleapis.com:443/v1/traces",
+            )
+        )
+    )
+    trace.set_tracer_provider(tracer_provider)
+
+    logger_provider = LoggerProvider(resource=resource)
+    logger_provider.add_log_record_processor(
+        BatchLogRecordProcessor(CloudLoggingExporter())
+    )
+    logs.set_logger_provider(logger_provider)
+
+    event_logger_provider = EventLoggerProvider(logger_provider)
+    events.set_event_logger_provider(event_logger_provider)
+
+    reader = PeriodicExportingMetricReader(CloudMonitoringMetricsExporter())
+    meter_provider = MeterProvider(metric_readers=[reader], resource=resource)
+    metrics.set_meter_provider(meter_provider)
+
+    # Load instrumentors
+    SQLite3Instrumentor().instrument()
+    VertexAIInstrumentor().instrument()
+    GoogleGenAiSdkInstrumentor().instrument()
+
+
+# [END opentelemetry_adk_otel_setup]
+
+
+def main() -> None:
+    # Make sure to set:
+    # OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED=true
+    # OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT=true
+    # in order to full prompts and responses and logs messages.
+    # For this sample, these can be set by loading the `main.env` file.
+    setup_opentelemetry()
+
+    # Call the function to get the FastAPI app instance.
+    # Ensure that the agent director name is the name of directory containing agent subdirectories,
+    # where each subdirectory represents a single agent with __init__.py and agent.py files.
+    # For this example this would be the current directory containing main.py.
+    # Note: Calling this method attempts to set the global tracer provider, which has already been
+    # set by the setup_opentelemetry() function.
+    app: FastAPI = get_fast_api_app(
```

Comment:
nit I think this was copypasta
```suggestion
    app = get_fast_api_app(
```